From 9c131cf3df4bedbfa36e4f8b743a96ca197c283d Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Fri, 21 Oct 2022 10:11:49 +0200
Subject: [PATCH 0525/1695] plat-stm32mp1: psci: use the stm32_system_reset

Call the stm32_system_reset function helper.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 662437883eb0ddc1c41db160b7ad1d3501bf81cc plat-stm32mp1: psci: use the stm32_system_reset]
Change-Id: Ieb0d202dead54ae8fbb7adc39b2b3c8ddd315e5a
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/358368
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/pm/psci.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp1/pm/psci.c b/core/arch/arm/plat-stm32mp1/pm/psci.c
index bb206808d..0ec3e267e 100644
--- a/core/arch/arm/plat-stm32mp1/pm/psci.c
+++ b/core/arch/arm/plat-stm32mp1/pm/psci.c
@@ -242,7 +242,8 @@ void __noreturn psci_system_off(void)
 /* Override default psci_system_reset() with platform specific sequence */
 void __noreturn psci_system_reset(void)
 {
-	rstctrl_assert(stm32mp_rcc_reset_id_to_rstctrl(MPSYST_R));
+	DMSG("core %u", get_core_pos());
+	stm32_reset_system();
 	udelay(100);
 	panic();
 }
-- 
2.47.2

